R/DataSet.R

```diff
@@ -26,66 +26,85 @@
-    } else if (format == TWO_COL) {
-      dat <- read_dat(datFile, subsampling)
+    
+    if (is.null(maximization)) {
+      maximization <- info$maximization
+      if (is.null(maximization) && !is.null(suite)) {
+        warning("maximization or minimization not specified in .info-file, 
+                 taking best guess based on the suite-name.")
+        if (grepl("\\w*bbob\\w*", suite, ignore.case = T) != 0) {
+          maximization <- F
+        }
+        else
+          maximization <- T
+      }
     }
+    
+    datBaseName <- strsplit(basename(info$datafile), '\\.')[[1]][1]
+    
+    datFile <- file.path(path, paste0(datBaseName, '.dat'))
+    tdatFile <- file.path(path, paste0(datBaseName, '.tdat'))
+    cdatFile <- file.path(path, paste0(datBaseName, '.cdat'))
+    
+
+    if (file.exists(cdatFile)) {
+      fv_file <- cdatFile
+    }
+    else if (file.exists(tdatFile)) {
+      fv_file <- tdatFile
+    }
+    else if (file.exists(datFile)) {
+      fv_file <- datFile
+    }
+    else {
+      return(NULL)
+    }
+    
+    if (file.exists(datFile)) {
+      rt_file <- datFile
+    }
+    else if (file.exists(cdatFile)) {
+      rt_file <- cdatFile
+    }
+    else if (file.exists(tdatFile)) {
+      rt_file <- tdatFile
+    }
+    
+    
+    read_unaligned <- switch(format,
+                             IOHprofiler = read_dat,
+                             COCO = read_COCO_dat2,
+                             BIOBJ_COCO = read_BIOBJ_COCO_dat,
+                             TWO_COL = read_dat)
+
+    fv_unaligned <- read_unaligned(fv_file, subsampling)
+    rt_unaligned <- read_unaligned(rt_file, subsampling)
+
+    if (is.null(maximization)) {
+      warning("Did not find maximization / minimization, auto-detecting based on
+               function value progression")
+      maximization <- (fv_unaligned[[1]][[1,2]] > 
+                       fv_unaligned[[1]][[1, length(fv_unaligned[[1]][1, ])]])
 
-    # check if the number of instances does not match
-    if (format != TWO_COL)
-      stopifnot(length(dat) == length(cdat))
-
-    if (is.null(maximization)){
-      if (format == IOHprofiler || format == TWO_COL)
-        maximization <- TRUE
-      else if (format == COCO)
-        maximization <- FALSE
     }
-    # RT <- align_by_target(dat, maximization = maximization, format = format) # runtime
-    RT <- align_runtime(dat, format = format, maximization = maximization)
+    
+    RT <- align_runtime(rt_unaligned, format = format, maximization = maximization)
+    FV <- align_function_value(fv_unaligned, format = format)  # function value
 
-    #TODO: check if this works correctly without CDAT-file
-    if (format == TWO_COL)
-      FV <- align_function_value(dat, format = format)  # function value
-    else
-      FV <- align_function_value(cdat, format = format)  # function value
 
     # TODO: remove this and incorporate the parameters aligned by runtimes
     FV[names(FV) != 'FV'] <- NULL
```

Comment:
I think we could just use the parameter aligned by FV instead, since it has a higher resolution?